### PR TITLE
refactor rest api endpoints to consolidate some shared resources

### PR DESCRIPTION
**Pull request by rolfyone**
Split out parameter check function into a library

Split out commonly used string resources

Addresses #1224

Signed-off-by: Paul Harris <paul.harris@consensys.net>

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java [hunk 1]
```diff
@@ -49,15 +53,16 @@ public BeaconChainHeadHandler(ChainStorageClient client, JsonProvider jsonProvid
...
       responses = {
         @OpenApiResponse(
-            status = "200",
+            status = RES_OK,
             content = @OpenApiContent(from = BeaconChainHeadResponse.class)),
         @OpenApiResponse(
-            status = "204",
-            description = "No Content will be returned if pre Genesis state")
+            status = RES_NO_CONTENT,
+            description = "No Content will be returned if pre Genesis state"),
```
Comment (macfarla) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java]: can we use NO_CONTENT_PRE_GENESIS

**Review by macfarla — APPROVED** [review]
LGTM - one nit

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/RestApiUtils.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   *
+   * @param parameterMap
+   * @param key
+   * @return returns the Value from the key, or throws IllegalArgumentException
+   */
+  public static String validateQueryParameter(
+      final Map<String, List<String>> parameterMap, final String key)
+      throws IllegalArgumentException {
+    if (parameterMap.containsKey(key)
+        && parameterMap.get(key).size() == 1
```
Comment (mark-terry) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/RestApiUtils.java:33]: It's doing `paramaterMap.get(key)` at least twice, perhaps local var?

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java (hunk 1 above)
Comment (mark-terry) [reply data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java]: +1, seems like a spot to use that?

**Review by mark-terry — COMMENTED** [review]
Some comments.

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconChainHeadHandler.java]: yep that was the intent :/ ta

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/RestApiUtils.java (hunk 2 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/RestApiUtils.java:33]: it'll be a lot more code, but sure, we could do that.

**Pull request merged**